### PR TITLE
Machines-integration-tests

### PR DESCRIPTION
**Pull request by maxiemwagenaar**
Add some unhappier tests

### CODE REVIEW

File: server/src/main/java/com/example/controller/MachineController.java [hunk 1]
```diff
@@ -18,7 +21,11 @@
 
 	@GetMapping("/{id}")
 	public Machine getMachineById(@PathVariable String id) {
-		return this.machineService.getOneById(id);
+		Optional<Machine> optionalMachine = this.machineService.getOneById(id);
+		if (optionalMachine.isPresent()) {
+			return optionalMachine.get();
+		}
+		throw new ServiceException("Could not find the machineId in the database.");
```
Comment (maxiemwagenaar) [inline server/src/main/java/com/example/controller/MachineController.java]: Should these be done like this or with something like an enum?

File: server/src/main/java/com/example/service/ProfileService.java [hunk 2]
```diff
@@ -57,24 +58,32 @@ public ProfileDto save(String id, EditProfile updateProfile) {
...
-		return null;
+		throw new ServiceException("Profile with this ID does not exist in the database.");
 	}
 
 	public Set<Machine> getAllMachines(String id) {
 		Optional<Profile> profile = profileRepository.findById(id);
 		Optional<Set<Machine>> machines = profile.map(Profile::getMachines);
-
-		return machines.orElse(Collections.emptySet());
+		if (profile.isPresent() && machines.isPresent()) {
```
Comment (mischaboldy) [inline server/src/main/java/com/example/service/ProfileService.java]: please add blank lines for ifs and returns

File: server/src/main/java/com/example/service/MachineService.java [hunk 3]
```diff
@@ -20,7 +21,8 @@
 
 	public Machine getOneById(String id) {
 		return machineRepository.findById(id)
-		                        .orElse(null);
+		                        .orElseThrow(() ->
+				                        new MissingMachineException(id));
```
Comment (jpastena) [inline server/src/main/java/com/example/service/MachineService.java]: formatting

File: server/src/main/java/com/example/service/ProfileService.java [hunk 4]
```diff
@@ -37,71 +37,91 @@ public Profile addOne(NewProfileDto newProfileDto) {
...
+		                 .orElseThrow(() -> new MissingProfileException(id));
+	}
+
+	private Machine findMachineById(String id) {
+		return machineRepository.findById(id)
+		                        .orElseThrow(() -> new MissingMachineException(id));
+	}
+
+	private List<String> getMachineIdDifference(List<Machine> machines, List<String> machineIds) {
+		for (Machine machine : machines) {
```
Comment (jpastena) [inline server/src/main/java/com/example/service/ProfileService.java]: let's try to use stream for this one

File: server/src/test/java/com/example/demo/integration/MachineControllerTest.java [hunk 5]
```diff
@@ -21,10 +26,7 @@
 import org.springframework.web.context.WebApplicationContext;
 import org.springframework.mock.web.MockMultipartFile;
 
-import java.time.ZoneId;
-import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
-
+import static org.hamcrest.CoreMatchers.isA;
```
Comment (jpastena) [inline server/src/test/java/com/example/demo/integration/MachineControllerTest.java]: this worry me a bit

File: server/src/test/java/com/example/demo/integration/ProfileControllerTest.java [hunk 6]
```diff
@@ -165,6 +180,52 @@ public void updateProfileShouldUpdateExistingProfile() throws Exception {
...
+
+		EditProfile editProfile = new EditProfile();
+		editProfile.setName(newName);
+		editProfile.setDescription(newDescription);
+
+		RequestBuilder requestBuilder = put("/profiles/{id}", PROFILE_0_ID).content(jsonUtil.asJson(editProfile))
+		                                                                   .contentType(MediaType.APPLICATION_JSON)
+		                                                                   .accept(MediaType.APPLICATION_JSON);
+		mockMvc.perform(requestBuilder)
+					 .andDo(printResponse())
```
[the remark this answers is not in the document]
Comment (jpastena) [reply server/src/test/java/com/example/demo/integration/ProfileControllerTest.java]: github has different tab size, that is not easily configurable, so the indentation looks strange

File: server/src/test/java/com/example/demo/integration/ProfileControllerTest.java [hunk 7]
```diff
@@ -165,6 +181,52 @@ public void updateProfileShouldUpdateExistingProfile() throws Exception {
...
+		                                                                   .contentType(MediaType.APPLICATION_JSON)
+		                                                                   .accept(MediaType.APPLICATION_JSON);
+
+		mockMvc.perform(requestBuilder)
+		       .andDo(printResponse())
+		       .andExpect(status().isBadRequest());
+	}
+
+	@Test
+	public void updateProfileWithWrongIdShouldMissingProfileException() throws Exception {
```
Comment (jpastena) [inline server/src/test/java/com/example/demo/integration/ProfileControllerTest.java]: should-throw

**Pull request merged**